certora/specs/LastUpdated.spec

```diff
@@ -0,0 +1,33 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+import "Enabled.spec";
+
+using MorphoHarness as Morpho;
+
+methods {
+    function Morpho.lastUpdate(MorphoHarness.Id) external returns(uint256) envfree;
+    function Morpho.libId(MorphoHarness.MarketParams) external returns(MorphoHarness.Id) envfree;
+}
+
+// Check that any positive cap market must come from a market that has been created on Morpho Blue.
+// The corresponding invariant cannot be verified because it requires to check properties on MetaMorpho and on Blue at the same time:
+// - on MetaMorpho, that it holds when the cap is positive for the first time
+// - on Blue, that a created market always has positive last update
+function hasPositiveSupplyCapIsUpdated(MetaMorphoHarness.Id id) returns bool {
+    uint192 supplyCap;
+    supplyCap, _, _ = config(id);
+
+    return supplyCap > 0 => Morpho.lastUpdate(id) > 0;
+}
+
+// Check that any new market in the supply queue necessarily has a positive cap.
+rule newSupplyQueueEnsuresPositiveCap(env e, MetaMorphoHarness.Id[] newSupplyQueue) {
+    uint256 i;
+
+    setSupplyQueue(e, newSupplyQueue);
+
+    MetaMorphoHarness.Id id = supplyQueue(i);
+
+    uint192 supplyCap;
+    supplyCap, _, _ = config(id);
+    assert supplyCap > 0;
+}
```

Comment:
I mean it makes sense to check supply cap > 0 for items in the supply queue, otherwise what's the point of putting it in the supply queue ?

Also it is preferable if you have only enabled markets in the supply queue, to avoid having interactions with non-authorized markets. Given that, checking supply cap > 0 instead does not seem to cost much

Reply:
My point was that this check doesn't ensure any invariant, as you can put the cap at zero after having put the market in the supply queue, and even disable it.

Reply:
You ensure that markets in the supply queue have been enabled at some point. I agree that it's not as strong as it would be ideally, but at least those markets kind of have been trusted at some point (so maybe they have been verified to not have a backdoor). Also it ensures that those use a consistent asset

Reply:
I say trusted because they have gone through a timelock, so we could imagine the guardian (or anyone else) having checked those markets in depth during that time